embedded/embedded-impl/pom.xml

```diff
@@ -23,6 +23,9 @@
         <javaModuleName>org.mule.distribution.embedded.impl</javaModuleName>
         <skipExportTests>false</skipExportTests>
         <formatterConfigPath>../../formatter.xml</formatterConfigPath>
+        <java17SourceDirectory>src/main/java17</java17SourceDirectory>
```

Comment:
The MRJAR is needed to avoid having to open the package `org.mule.runtime.module.embedded.internal.controller`. This way the `module-info` can be only for Java 17, where we will actually be using module layers.